src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ReachabilityInstrumentationProvider.cs

```diff
@@ -0,0 +1,310 @@
+                    builder.EmitBytes(mvid.ToByteArray());
+                    builder.EmitInt(item.Value.Length);
+
+                    foreach (ExternSymbolMappedField field in item.Value)
+                    {
+                        if (field != null)
+                        {
+                            builder.AddSymbol(new UntrackedSymbol(builder.CountBytes, field.SymbolName));
+                        }
+                        builder.EmitByte(0);
+                    }
+                }
+
+                builder.EmitInt(lengthReservation, builder.CountBytes);
+
+                return builder.ToObjectData();
+            }
+
+            private sealed class UntrackedSymbol : ISymbolDefinitionNode
+            {
+                private readonly int _offset;
+                private readonly string _name;
+
+                public UntrackedSymbol(int offset, string name)
+                    => (_offset, _name) = (offset, name);
+
+                int ISymbolDefinitionNode.Offset => _offset;
+                int ISymbolNode.Offset => 0;
+
+                bool ISymbolNode.RepresentsIndirectionCell => false;
+                bool IDependencyNode<NodeFactory>.InterestingForDynamicDependencyAnalysis => false;
+                bool IDependencyNode<NodeFactory>.HasDynamicDependencies => false;
+                bool IDependencyNode<NodeFactory>.HasConditionalStaticDependencies => false;
+                bool IDependencyNode<NodeFactory>.StaticDependenciesAreComputed => true;
+                bool IDependencyNode.Marked => true;
+
+                void ISymbolNode.AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb) => sb.Append(_name);
+                IEnumerable<CombinedDependencyListEntry> IDependencyNode<NodeFactory>.GetConditionalStaticDependencies(NodeFactory context) => throw new NotImplementedException();
+                IEnumerable<DependencyListEntry> IDependencyNode<NodeFactory>.GetStaticDependencies(NodeFactory context) => null;
+                IEnumerable<CombinedDependencyListEntry> IDependencyNode<NodeFactory>.SearchDynamicDependencies(List<DependencyNodeCore<NodeFactory>> markedNodes, int firstNode, NodeFactory context) => throw new NotImplementedException();
+            }
+        }
+
+        private sealed partial class InitializeMethod : ILStubMethod
+        {
+            private readonly CompilerTypeSystemContext _context;
+            private readonly ExternSymbolMappedField _dataField;
+            private MethodSignature _signature;
+
+            public InitializeMethod(CompilerTypeSystemContext context)
+            {
+                _context = context;
+                _dataField = new ExternSymbolMappedField(context.GetWellKnownType(WellKnownType.Int32), TokenUseBeginSymbol);
+            }
+
+            public override TypeSystemContext Context
+            {
+                get
+                {
+                    return _context;
+                }
+            }
```

Comment:
FYI you could use an auto-property to make these shorter if you want, e.g.

```suggestion
            public override TypeSystemContext Context { get; }
```